### PR TITLE
UI/transit auto rotate interval

### PR DESCRIPTION
**Pull request by hashishaw**
Add support in the UI for the `auto_rotate_interval` attribute on transit keys. 

**CREATE: on page load auto rotate interval is off (0) by default**
Create key - no rotate by default

**CREATE: Enabling auto rotate sets the initial value to 1hr, which is the minimum**
Create key - 1 hr default

**EDIT: Validation ensures that the value will not be under 1 hour**
Validation under 1 hr

**SHOW: Parses the duration as time string**
Show parsed duration

**When value is 0s, show default text**
Show Key will not be rotated

### CODE REVIEW

File: ui/app/helpers/format-ttl.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { helper } from '@ember/component/helper';
+
+export default helper(function formatTtl([timestring], { removeZero = false }) {
+  // Expects a number followed by one of s, m, or h
+  // eg. 40m or 1h20m0s
+  let matches = timestring?.match(/([0-9]+[h|m|s])/g);
```
Comment (Monkeychip) [inline ui/app/helpers/format-ttl.js:6]: nice regex work

Comment (hashishaw) [reply ui/app/helpers/format-ttl.js:6]: Thank you! Took me way longer than I'd like to admit 🙈 😂

File: ui/app/templates/components/transit-form-create.hbs [hunk 2]
```diff
@@ -6,6 +6,17 @@
...
     </div>
+    <div class="field">
+      <TtlPicker2
+        @initialValue="1h"
+        @initialEnabled={{false}}
+        @label="Auto-rotation interval"
+        @helperTextDisabled="Key will never be automatically rotated"
+        @helperTextEnabled="Key will be automatically rotated every"
+        @onChange={{@handleAutoRotateChange}}
+        @errorMessage={{if @autoRotateInvalid "Duration must be longer than 1 hour"}}
```
Comment (Monkeychip) [inline ui/app/templates/components/transit-form-create.hbs:17]: love the validation. thank you for adding that!

File: ui/tests/integration/components/ttl-picker2-test.js [hunk 3]
```diff
@@ -7,9 +7,11 @@ import hbs from 'htmlbars-inline-precompile';
 module('Integration | Component | ttl-picker2', function (hooks) {
   setupRenderingTest(hooks);
 
+  hooks.beforeEach(function () {
```
Comment (Monkeychip) [inline ui/tests/integration/components/ttl-picker2-test.js:10]: nice job on pulling this out and cleaning things up

File: ui/tests/integration/helpers/format-ttl-test.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import { module, test } from 'qunit';
```
Comment (Monkeychip) [inline ui/tests/integration/helpers/format-ttl-test.js:1]: nice test coverage!

**Review by Monkeychip — APPROVED** [review]
Great work!

**Pull request merged**